x-pack/legacy/plugins/canvas/server/usage/workpad_collector.ts

```diff
@@ -87,7 +73,7 @@ export function summarizeWorkpads(workpadDocs: Workpad[]): WorkpadTelemetry {
     );
     const functionCounts = workpad.pages.reduce<number[]>((accum, page) => {
       return page.elements.map(element => {
-        const ast: AST = fromExpression(element.expression) as AST; // TODO: Remove once fromExpression is properly typed
+        const ast: ExpressionAST = fromExpression(element.expression) as ExpressionAST; // TODO: Remove once fromExpression is properly typed
```

Comment:
@ppisljar @lukeelmers : will `fromExpression` be properly typed anytime soon?

Reply:
@ppisljar Any reason not to go ahead and type it as this ExpressionAST? I can add that to this PR if needed?

Reply:
@crob611 The `ExpressionAST` type should be safe to use here.

@clintandrewhall It looks like a type for `fromExpression` is already defined in the old interpreter package (`packages/kbn-interpreter/src/common/lib/ast.d.ts`), but it uses `AST: unknown` (this was before we added typings for the AST). This section must have been overlooked when we made that update; I'll make a note to fix it, and can remove the casting you've done here at that time.

Reply:
(or @crob611 if you want to make that change in this PR feel free to do that as well!)

Reply:
(sorry for all the comments)

The issue you are going to run into is that the interface for the AST is defined in the data plugin, which means you can't import it into the old interpreter package. So it may be better to remove the `fromExpression` type from the interpreter package, and move it to the data plugin to export from there.

Reply:
@lukeelmers How would you set the type for something in kbn/interpreter from within the data plugin? Would you have to use like `declare module 'kbn-interpreter`?